targets/UnrealMarketplacePlugin/examplesource/ExampleProject/Plugins/PlayFab/Source/PlayFabCommon/Private/TestFramework/PlayFabTestRunner.cpp.ejs

```diff
@@ -16,10 +18,90 @@ UPlayFabTestRunner::UPlayFabTestRunner(const FObjectInitializer& ObjectInitializ
 {
 }
 
+UTestTitleDataLoader::UTestTitleDataLoader()
+{
+    isValid = LoadTestTitleData();
+}
+
+bool UTestTitleDataLoader::LoadFromCode()
+{
+    // README:
+    //   In order for tests to run, you need a valid testTitleData.json file
+    //   Format specification here: https://github.com/PlayFab/SDKGenerator/blob/master/JenkinsConsoleUtility/testTitleData.md
+    // Option 1: You can drop a testTitleData.json file into this location in your Unreal Project:
+    //   FPaths::ProjectContentDir() + TEXT("/TestTitleData/testTitleData.json");
+    // Option 2: You can specify the PF_TEST_TITLE_DATA_JSON environment variable
+    //   (Windows or Linux), containing the full path of a testTitleData.json file, anywhere on your computer
+    // Option 3:
+    //   Delete this "return false;", and fill in the code variables below with your title's information
+
+    return false;
+
+    titleId = "<your title id>";
+    developerSecretKey = "<your title secret>";
+    userEmail = "<a valid email linked to LoginWithEmailAddress>";
+
+    return true;
+}
+
+bool UTestTitleDataLoader::LoadFromFile()
+{
+    bool success = true;
+
+    FString jsonInput;
+    FString filename = FPaths::ProjectContentDir() + TEXT("/TestTitleData/testTitleData.json");
+
+    if (!FPaths::FileExists(filename))
+    {
+        // Prefer to load path from environment variable, if present
+#if PLATFORM_WINDOWS
+        char* envPath = nullptr;
+        size_t envPathStrLen;
+        errno_t err = _dupenv_s(&envPath, &envPathStrLen, "PF_TEST_TITLE_DATA_JSON");
+        if (err == 0 && envPath != nullptr)
+            filename = FString(ANSI_TO_TCHAR(envPath));
+        if (envPath != nullptr)
+            free(envPath);
+#endif
+    }
+
+    FString titleDataJson;
+    success &= FPaths::FileExists(filename) && FFileHelper::LoadFileToString(titleDataJson, *filename);
```

Comment:
>success [](start = 4, length = 7)

success is not used until this point.  consider declaring the variable here instead of above.